src/app/pages/auth/components/qr-form/qr-form.component.ts

```diff
@@ -13,13 +13,23 @@ export class QRFormComponent {
   @Output()
   data: EventEmitter<any> = new EventEmitter<any>()
 
-  constructor(private scanner: BarcodeScanner) {}
+  constructor() {}
 
-  scanQRHandler() {
-    const scanOptions = {
-      showFlipCameraButton: true,
-      orientation: 'portrait'
-    }
-    return this.scanner.scan(scanOptions).then(res => this.data.emit(res.text))
+  async scanQRHandler() {
+    document.querySelector('body').classList.add('scanner-active')
+    // Check camera permission
+    // This is just a simple example, check out the better checks below
+    await BarcodeScanner.requestPermissions()
+
+     // Add the `barcodeScanned` listener
+    const listener = await BarcodeScanner.addListener(
+    'barcodeScanned',
+    async result => {
+      await listener.remove();
+      this.data.emit(result.barcode.rawValue)
+      document.querySelector('body').classList.remove('scanner-active');
```

Comment:
Add comment: "Remove this when plugin is fixed" since this is only a workaround.